### ISSUE TITLE
option not to load the entire repository, but only selected files

### ISSUE THREAD

**Issue by pjhfggij** [orgzly-revived/orgzly-android-revived#217]
- I have searched for existing issues that may be the same as or related to mine.

## the issue
I've tried to add a single file to orgzly with a link from a folder that has a lot of other .org files. This folder is synced (via [RSAF](https://github.com/chenxiaolong/RSAF)) from my desktop's ~/notes folder. Upon adding this folder to repositories and pressing sync orgzly loaded all files in this folder even though I wanted to choose only a few of them.

To make matters worse you can't multiselect notebooks to delete them and you'd have to do it one by one (and it prompts you each time "are you sure you want to delete") forcing me to reinstall orgzly and lose all customizations to clean this up.

## the proposed solution
add an option not to sync the entire repository, but only files that the user sets a link to

## what I've tried
Android Storage Scopes, but they only work on local directories and not on Storage Frameworks like RSAF

**Comment by amberin** [comment]
Interesting suggestion. How do you propose the UI should work? I mean, how would you specify which files to sync, and which ones to ignore?

I'm thinking this sort of clashes with the "repository" concept, so I imagine the UX story would need to look quite different.

**Comment by pjhfggij** [comment]
@amberin what if we added an option to Settings > Sync > "Sync only notebooks with a link"

If for any reason there was a bug that would accidentally sync all the notebooks into orgzly, I'd appreciate also a function to "unload all notebooks without a link". As mentioned it would have taken me hours to remove all the accidentally synced notebooks.

**Comment by pjhfggij** [comment]
You are right, it is more involved. How about instead of adding this "Sync only notebooks with a link" option there was an entry in the "three dots" menu called "Link Org file" (after "Import Org file")
![image](https://github.com/orgzly-revived/orgzly-android-revived/assets/161205138/fbcfa57a-df33-4c9d-9ec4-e6fb4ac7be79)

that would import, set link and create a new repository (without importing any other files) all in one go?

**Comment by daraul** [comment]
What about letting one edit an `.orgzlyignore` file in the repository? As for the UI, it'd just be an item somewhere in the repository settings menu that opens up a text editor for the file. Upon save, sync that repository, and remove any files that are in the ignore list -- be nice if the ignore list worked similar to [git's](https://git-scm.com/docs/gitignore). 

Then you could do that @pjhfggij wants to with an ignore file like this: 

```
!that-one-file.org
```

**Comment by amberin** [comment]
The `.orgzlyignore` file is already exists as a feature when using the Git repo type. I have plans to make it available for all repo types, but I can't seem to find that issue right now.

I'm hoping that would be good enough as a substitute for this feature, but I can also imagine scenarios where it would be impractical. For instance, repos with a lot of changes where files come and go all the time.

**Comment by daraul** [comment]
Odd, I don't see git as a repo type. Do I need to do something to access it? 

![Screenshot_20240420_060109_Orgzly Revived](https://github.com/orgzly-revived/orgzly-android-revived/assets/2440975/e17a1f34-000d-42cb-bc63-f2e9ac71b45f)

**Comment by amberin** [comment]
Yes, it is still an experimental feature, so you need to enable it in Settings -> App -> Developer settings.

**Comment by pjhfggij** [comment]
> What about letting one edit an `.orgzlyignore` file in the repository? As for the UI, it'd just be an item somewhere in the repository settings menu that opens up a text editor for the file. Upon save, sync that repository, and remove any files that are in the ignore list -- be nice if the ignore list worked similar to [git's](https://git-scm.com/docs/gitignore).
> 
> Then you could do what @pjhfggij wants to with an ignore file like this:
> 
> ```
> !that-one-file.org
> ```

I think that's a great idea, if it worked for all repo types that would work for me

**Comment by amberin** [comment]
> It would be useful if orgzly could unload notebook that are excluded in this file as well if they were imported before (as mentioned it's really hard right now to delete multiple notebooks via the default interface).

@pjhfggij The current behavior of the ignore file (in Git repos) is this: If a new ignore rule is added which matches an existing book in the repo, then the book's repo link is removed upon the next sync, and an error is raised. I believe simply deleting the book from Orgzly would be a bit reckless/risky.

I feel your pain about deleting multiple notebooks, but that is a separate, long-standing issue which I think needs fixing, anyway. I have long been missing the ability to select multiple notebooks, and I haven't looked into it, but it really should not be hard to fix.

**Comment by pjhfggij** [comment]
makes sense, I've seen that you've opened it as a [separate issue already](https://github.com/orgzly-revived/orgzly-android-revived/issues/227) as well. Hopefully, it will be an easy fix

**Comment by amberin** [comment]
This should be solved in v1.8.26. If there are issues with the .orgzlyignore file, let's handle them separately.

**Comment by pjhfggij** [comment]
fantastic! thank you! as soon as 1.8.26 reaches fdroid I will report if this .orgzlyignore works for me:
```gitignore
# ignore everything by default
*

# except for "test.org"
!test.org

# except for files in folder "orgzly"
!orgzly/
!orgzly/*
```
